// ngx-content-loading/bundles/ngx-content-loading.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define('ngx-content-loading', ['@angular/core','@angular/common','exports', '@angular/core', '@angular/common'], factory) :
	(factory(global.ng.core,global.ng.common,(global['ngx-content-loading'] = {}),global.ng.core,global.ng.common));
}(this, (function (ɵngcc0,ɵngcc1,exports,core,common) { 
var _c0 = [[["", "before-svg", ""]], "*", [["", "after-svg", ""]]];
var _c1 = ["[before-svg]", "*", "[after-svg]"];
var _c2 = ["ngx-rect", ""];
var _c3 = ["ngx-svg-stop", ""];
var _c4 = ["ngx-facebook-preset", ""];
var _c5 = ["ngx-instagram-preset", ""];
var _c6 = ["ngx-circle", ""];
var _c7 = ["ngx-code-preset", ""];
var _c8 = ["ngx-bullet-list-preset", ""];
var _c9 = ["ngx-list-preset", ""];
'use strict';

/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0
THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.
See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */
/* global Reflect, Promise */
var extendStatics = Object.setPrototypeOf ||
    ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
    function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
function __extends(d, b) {
    extendStatics(d, b);
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
}

var Defaults = {
    PrimaryColor: '#f3f3f3',
    SecondaryColor: '#ecebeb',
    PreserveAspectRatio: 'xMidYMid meet',
    Classes: '',
    Speed: '1000ms',
    Width: '400',
    Height: '130',
    Rx: '0',
    Ry: '1',
};
var NgxContentLoadingComponent = /** @class */ (function () {
    function NgxContentLoadingComponent() {
        this.width = Defaults.Width;
        this.height = Defaults.Height;
        this.primaryColor = Defaults.PrimaryColor;
        this.secondaryColor = Defaults.SecondaryColor;
        this.speed = Defaults.Speed;
        this.preserveAspectRatio = Defaults.PreserveAspectRatio;
    }
    Object.defineProperty(NgxContentLoadingComponent.prototype, "viewBox", {
        get: function () {
            return "0 0 " + this.width + " " + this.height;
        },
        enumerable: true,
        configurable: true
    });
    NgxContentLoadingComponent.prototype.ngOnInit = function () { };
NgxContentLoadingComponent.ɵfac = function NgxContentLoadingComponent_Factory(t) { return new (t || NgxContentLoadingComponent)(); };
NgxContentLoadingComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgxContentLoadingComponent, selectors: [["ngx-content-loading"]], inputs: { width: "width", height: "height", primaryColor: "primaryColor", secondaryColor: "secondaryColor", speed: "speed", preserveAspectRatio: "preserveAspectRatio" }, ngContentSelectors: _c1, decls: 4, vars: 2, consts: [["version", "1.1"]], template: function NgxContentLoadingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵprojection(0);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(1, "svg", 0);
        ɵngcc0.ɵɵprojection(2, 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(3, 2);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("preserveAspectRatio", ctx.preserveAspectRatio)("viewBox", ctx.viewBox);
    } }, styles: [""], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxContentLoadingComponent, [{
        type: core.Component,
        args: [{
                selector: 'ngx-content-loading',
                template: "<ng-content select=\"[before-svg]\"></ng-content>\n<svg version=\"1.1\"\n    [attr.preserveAspectRatio]=\"preserveAspectRatio\"\n    [attr.viewBox]=\"viewBox\">\n        <ng-content></ng-content>\n</svg>\n<ng-content select=\"[after-svg]\"></ng-content>\n",
                styles: [""],
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, { width: [{
            type: core.Input
        }], height: [{
            type: core.Input
        }], primaryColor: [{
            type: core.Input
        }], secondaryColor: [{
            type: core.Input
        }], speed: [{
            type: core.Input
        }], preserveAspectRatio: [{
            type: core.Input
        }] }); })();
    return NgxContentLoadingComponent;
}());
NgxContentLoadingComponent.ctorParameters = function () { return []; };
NgxContentLoadingComponent.propDecorators = {
    "width": [{ type: core.Input },],
    "height": [{ type: core.Input },],
    "primaryColor": [{ type: core.Input },],
    "secondaryColor": [{ type: core.Input },],
    "speed": [{ type: core.Input },],
    "preserveAspectRatio": [{ type: core.Input },],
};
var SvgElementComponent = /** @class */ (function () {
    function SvgElementComponent(content) {
        this.content = content;
        this.clipPathId = getId();
        this.linearGradientId = getId();
    }
    Object.defineProperty(SvgElementComponent.prototype, "clipPathUrl", {
        get: function () {
            return "url(#" + this.clipPathId + ")";
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SvgElementComponent.prototype, "linearGradientUrl", {
        get: function () {
            return "url(#" + this.linearGradientId + ")";
        },
        enumerable: true,
        configurable: true
    });
    SvgElementComponent.prototype.ngOnInit = function () {
    };
SvgElementComponent.ɵfac = function SvgElementComponent_Factory(t) { return new (t || SvgElementComponent)(ɵngcc0.ɵɵdirectiveInject(NgxContentLoadingComponent)); };
SvgElementComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SvgElementComponent, selectors: [["ngx-svg-element"]], decls: 0, vars: 0, template: function SvgElementComponent_Template(rf, ctx) { }, styles: [""], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SvgElementComponent, [{
        type: core.Component,
        args: [{
                selector: 'ngx-svg-element',
                template: "",
                styles: [""],
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return [{ type: NgxContentLoadingComponent }]; }, null); })();
    return SvgElementComponent;
}());
SvgElementComponent.ctorParameters = function () { return [
    { type: NgxContentLoadingComponent, },
]; };
function getId() {
    var id = performance.now() * Math.random();
    return id.toString();
}
var RectComponent = /** @class */ (function (_super) {
    __extends(RectComponent, _super);
    function RectComponent(content) {
        var _this = _super.call(this, content) || this;
        _this.content = content;
        _this.rx = Defaults.Rx;
        _this.ry = Defaults.Ry;
        return _this;
    }
RectComponent.ɵfac = function RectComponent_Factory(t) { return new (t || RectComponent)(ɵngcc0.ɵɵdirectiveInject(NgxContentLoadingComponent)); };
RectComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: RectComponent, selectors: [["", "ngx-rect", ""]], inputs: { y: "y", x: "x", rx: "rx", ry: "ry", width: "width", height: "height", primaryColor: "primaryColor", secondaryColor: "secondaryColor", speed: "speed" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], attrs: _c2, decls: 7, vars: 22, consts: [["ngx-svg-stop", "", 3, "offset", "values", "speed", "color"]], template: function RectComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "rect");
        ɵngcc0.ɵɵelementStart(1, "defs");
        ɵngcc0.ɵɵelement(2, "clipPath");
        ɵngcc0.ɵɵelementStart(3, "linearGradient");
        ɵngcc0.ɵɵelement(4, "stop", 0);
        ɵngcc0.ɵɵelement(5, "stop", 0);
        ɵngcc0.ɵɵelement(6, "stop", 0);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("x", ctx.x)("y", ctx.y)("rx", ctx.rx)("ry", ctx.ry)("width", ctx.width)("fill", ctx.linearGradientUrl)("clipPath", ctx.clipPathUrl)("height", ctx.height);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("id", ctx.clipPathId);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("id", ctx.linearGradientId);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "0%")("values", "-2; 1")("speed", ctx.content.speed)("color", ctx.content.primaryColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "50%")("values", "-1.5; 1.5")("speed", ctx.content.speed)("color", ctx.content.secondaryColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "100%")("values", "-1; 2")("speed", ctx.content.speed)("color", ctx.content.primaryColor);
    } }, directives: function () { return [SvgStopComponent]; }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RectComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-rect]',
                template: "<svg:rect\n        [attr.x]=\"x\"\n        [attr.y]=\"y\"\n        [attr.rx]=\"rx\"\n        [attr.ry]=\"ry\"\n        [attr.width]=\"width\"\n        [attr.fill]=\"linearGradientUrl\"\n        [attr.clipPath]=\"clipPathUrl\"\n        [attr.height]=\"height\">\n</svg:rect>\n<svg:defs>\n    <svg:clipPath [attr.id]=\"clipPathId\"></svg:clipPath>\n    <svg:linearGradient [attr.id]=\"linearGradientId\">\n        <svg:stop ngx-svg-stop\n            [offset]=\"'0%'\"\n            [values]=\"'-2; 1'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.primaryColor\">\n        </svg:stop>\n        <svg:stop ngx-svg-stop\n            [offset]=\"'50%'\"\n            [values]=\"'-1.5; 1.5'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.secondaryColor\">\n        </svg:stop>\n        <svg:stop ngx-svg-stop\n            [offset]=\"'100%'\"\n            [values]=\"'-1; 2'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.primaryColor\">\n        </svg:stop>\n    </svg:linearGradient>\n</svg:defs>\n",
                styles: [""]
            }]
    }], function () { return [{ type: NgxContentLoadingComponent }]; }, { y: [{
            type: core.Input
        }], x: [{
            type: core.Input
        }], rx: [{
            type: core.Input
        }], ry: [{
            type: core.Input
        }], width: [{
            type: core.Input
        }], height: [{
            type: core.Input
        }], primaryColor: [{
            type: core.Input
        }], secondaryColor: [{
            type: core.Input
        }], speed: [{
            type: core.Input
        }] }); })();
    return RectComponent;
}(SvgElementComponent));
RectComponent.ctorParameters = function () { return [
    { type: NgxContentLoadingComponent, },
]; };
RectComponent.propDecorators = {
    "y": [{ type: core.Input },],
    "x": [{ type: core.Input },],
    "rx": [{ type: core.Input },],
    "ry": [{ type: core.Input },],
    "width": [{ type: core.Input },],
    "height": [{ type: core.Input },],
    "primaryColor": [{ type: core.Input },],
    "secondaryColor": [{ type: core.Input },],
    "speed": [{ type: core.Input },],
};
var SvgStopComponent = /** @class */ (function () {
    function SvgStopComponent() {
    }
    SvgStopComponent.prototype.ngOnInit = function () {
    };
SvgStopComponent.ɵfac = function SvgStopComponent_Factory(t) { return new (t || SvgStopComponent)(); };
SvgStopComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SvgStopComponent, selectors: [["", "ngx-svg-stop", ""]], hostVars: 2, hostBindings: function SvgStopComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("offset", ctx.offset)("stop-color", ctx.color);
    } }, inputs: { speed: "speed", values: "values", offset: "offset", color: "color" }, attrs: _c3, decls: 1, vars: 2, consts: [["attributeName", "offset", "repeatCount", "indefinite"]], template: function SvgStopComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "animate", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("values", ctx.values)("dur", ctx.speed);
    } }, styles: [""], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SvgStopComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-svg-stop]',
                template: "<svg:animate\n    attributeName=\"offset\"\n    [attr.values]=\"values\"\n    [attr.dur]=\"speed\"\n    repeatCount=\"indefinite\"\n/>\n",
                styles: [""],
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, { speed: [{
            type: core.Input
        }], values: [{
            type: core.Input
        }], offset: [{
            type: core.HostBinding,
            args: ['attr.offset']
        }, {
            type: core.Input
        }], color: [{
            type: core.HostBinding,
            args: ['attr.stop-color']
        }, {
            type: core.Input
        }] }); })();
    return SvgStopComponent;
}());
SvgStopComponent.ctorParameters = function () { return []; };
SvgStopComponent.propDecorators = {
    "speed": [{ type: core.Input },],
    "values": [{ type: core.Input },],
    "offset": [{ type: core.HostBinding, args: ['attr.offset',] }, { type: core.Input },],
    "color": [{ type: core.HostBinding, args: ['attr.stop-color',] }, { type: core.Input },],
};
var FacebookPresetComponent = /** @class */ (function () {
    function FacebookPresetComponent() {
    }
    FacebookPresetComponent.prototype.ngOnInit = function () {
    };
FacebookPresetComponent.ɵfac = function FacebookPresetComponent_Factory(t) { return new (t || FacebookPresetComponent)(); };
FacebookPresetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FacebookPresetComponent, selectors: [["", "ngx-facebook-preset", ""]], attrs: _c4, decls: 6, vars: 0, consts: [["ngx-circle", "", "cx", "25", "cy", "25", "r", "25"], ["ngx-rect", "", "width", "100", "height", "5", "y", "15", "x", "60", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "60", "height", "5", "y", "30", "x", "60", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "170", "height", "5", "y", "65", "x", "0", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "220", "height", "5", "y", "80", "x", "0", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "130", "height", "5", "y", "95", "x", "0", "rx", "5", "ry", "5"]], template: function FacebookPresetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "g", 0);
        ɵngcc0.ɵɵelement(1, "g", 1);
        ɵngcc0.ɵɵelement(2, "g", 2);
        ɵngcc0.ɵɵelement(3, "g", 3);
        ɵngcc0.ɵɵelement(4, "g", 4);
        ɵngcc0.ɵɵelement(5, "g", 5);
    } }, directives: function () { return [CircleComponent, RectComponent]; }, styles: [""], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FacebookPresetComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-facebook-preset]',
                template: "<svg:g ngx-circle cx=\"25\" cy=\"25\" r=\"25\"></svg:g>\n<svg:g ngx-rect width=\"100\" height=\"5\" y=\"15\" x=\"60\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"60\" height=\"5\" y=\"30\" x=\"60\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"170\" height=\"5\" y=\"65\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"220\" height=\"5\" y=\"80\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"130\" height=\"5\" y=\"95\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n",
                styles: [""],
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, null); })();
    return FacebookPresetComponent;
}());
FacebookPresetComponent.ctorParameters = function () { return []; };
var InstagramPresetComponent = /** @class */ (function () {
    function InstagramPresetComponent() {
    }
    InstagramPresetComponent.prototype.ngOnInit = function () {
    };
InstagramPresetComponent.ɵfac = function InstagramPresetComponent_Factory(t) { return new (t || InstagramPresetComponent)(); };
InstagramPresetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InstagramPresetComponent, selectors: [["", "ngx-instagram-preset", ""]], attrs: _c5, decls: 4, vars: 0, consts: [["ngx-circle", "", "cx", "25", "cy", "25", "r", "25"], ["ngx-rect", "", "width", "120", "height", "10", "y", "10", "x", "60", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "40", "height", "5", "y", "30", "x", "60", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "250", "height", "250", "y", "60", "x", "0", "rx", "5", "ry", "5"]], template: function InstagramPresetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "g", 0);
        ɵngcc0.ɵɵelement(1, "g", 1);
        ɵngcc0.ɵɵelement(2, "g", 2);
        ɵngcc0.ɵɵelement(3, "g", 3);
    } }, directives: function () { return [CircleComponent, RectComponent]; }, styles: [""], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InstagramPresetComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-instagram-preset]',
                template: "<svg:g ngx-circle cx=\"25\" cy=\"25\" r=\"25\"></svg:g>\n<svg:g ngx-rect width=\"120\" height=\"10\" y=\"10\" x=\"60\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"40\" height=\"5\" y=\"30\" x=\"60\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"250\" height=\"250\" y=\"60\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n",
                styles: [""],
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, null); })();
    return InstagramPresetComponent;
}());
InstagramPresetComponent.ctorParameters = function () { return []; };
var CircleComponent = /** @class */ (function (_super) {
    __extends(CircleComponent, _super);
    function CircleComponent(content) {
        var _this = _super.call(this, content) || this;
        _this.content = content;
        return _this;
    }
CircleComponent.ɵfac = function CircleComponent_Factory(t) { return new (t || CircleComponent)(ɵngcc0.ɵɵdirectiveInject(NgxContentLoadingComponent)); };
CircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CircleComponent, selectors: [["", "ngx-circle", ""]], inputs: { r: "r", cx: "cx", cy: "cy" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], attrs: _c6, decls: 7, vars: 19, consts: [["ngx-svg-stop", "", 3, "offset", "values", "speed", "color"]], template: function CircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "circle");
        ɵngcc0.ɵɵelementStart(1, "defs");
        ɵngcc0.ɵɵelement(2, "clipPath");
        ɵngcc0.ɵɵelementStart(3, "linearGradient");
        ɵngcc0.ɵɵelement(4, "stop", 0);
        ɵngcc0.ɵɵelement(5, "stop", 0);
        ɵngcc0.ɵɵelement(6, "stop", 0);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("cx", ctx.cx)("cy", ctx.cy)("r", ctx.r)("fill", ctx.linearGradientUrl)("clipPath", ctx.clipPathUrl);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("id", ctx.clipPathId);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("id", ctx.linearGradientId);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "0%")("values", "-2; 1")("speed", ctx.content.speed)("color", ctx.content.primaryColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "50%")("values", "-1.5; 1.5")("speed", ctx.content.speed)("color", ctx.content.secondaryColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("offset", "100%")("values", "-1; 2")("speed", ctx.content.speed)("color", ctx.content.primaryColor);
    } }, directives: [SvgStopComponent], styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-circle]',
                template: "<svg:circle\n        [attr.cx]=\"cx\"\n        [attr.cy]=\"cy\"\n        [attr.r]=\"r\"\n        [attr.fill]=\"linearGradientUrl\"\n        [attr.clipPath]=\"clipPathUrl\">\n</svg:circle>\n<svg:defs>\n    <svg:clipPath [attr.id]=\"clipPathId\"></svg:clipPath>\n    <svg:linearGradient [attr.id]=\"linearGradientId\">\n        <svg:stop ngx-svg-stop\n            [offset]=\"'0%'\"\n            [values]=\"'-2; 1'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.primaryColor\">\n        </svg:stop>\n        <svg:stop ngx-svg-stop\n            [offset]=\"'50%'\"\n            [values]=\"'-1.5; 1.5'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.secondaryColor\">\n        </svg:stop>\n        <svg:stop ngx-svg-stop\n            [offset]=\"'100%'\"\n            [values]=\"'-1; 2'\"\n            [speed]=\"content.speed\"\n            [color]=\"content.primaryColor\">\n        </svg:stop>\n    </svg:linearGradient>\n</svg:defs>\n",
                styles: [""]
            }]
    }], function () { return [{ type: NgxContentLoadingComponent }]; }, { r: [{
            type: core.Input
        }], cx: [{
            type: core.Input
        }], cy: [{
            type: core.Input
        }] }); })();
    return CircleComponent;
}(SvgElementComponent));
CircleComponent.ctorParameters = function () { return [
    { type: NgxContentLoadingComponent, },
]; };
CircleComponent.propDecorators = {
    "r": [{ type: core.Input },],
    "cx": [{ type: core.Input },],
    "cy": [{ type: core.Input },],
};
var CodePresetComponent = /** @class */ (function () {
    function CodePresetComponent() {
    }
    CodePresetComponent.prototype.ngOnInit = function () {
    };
CodePresetComponent.ɵfac = function CodePresetComponent_Factory(t) { return new (t || CodePresetComponent)(); };
CodePresetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CodePresetComponent, selectors: [["", "ngx-code-preset", ""]], attrs: _c7, decls: 9, vars: 0, consts: [["ngx-rect", "", "width", "60", "height", "10", "y", "15", "x", "0", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "90", "height", "10", "y", "15", "x", "70", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "20", "height", "10", "y", "15", "x", "170", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "100", "height", "10", "y", "35", "x", "9", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "120", "height", "10", "y", "35", "x", "120", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "90", "height", "10", "y", "55", "x", "5", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "90", "height", "10", "y", "55", "x", "105", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "70", "height", "10", "y", "55", "x", "205", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "30", "height", "10", "y", "75", "x", "0", "rx", "5", "ry", "5"]], template: function CodePresetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "g", 0);
        ɵngcc0.ɵɵelement(1, "g", 1);
        ɵngcc0.ɵɵelement(2, "g", 2);
        ɵngcc0.ɵɵelement(3, "g", 3);
        ɵngcc0.ɵɵelement(4, "g", 4);
        ɵngcc0.ɵɵelement(5, "g", 5);
        ɵngcc0.ɵɵelement(6, "g", 6);
        ɵngcc0.ɵɵelement(7, "g", 7);
        ɵngcc0.ɵɵelement(8, "g", 8);
    } }, directives: [RectComponent], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CodePresetComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-code-preset]',
                template: "<svg:g ngx-rect width=\"60\" height=\"10\" y=\"15\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"90\" height=\"10\" y=\"15\" x=\"70\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"20\" height=\"10\" y=\"15\" x=\"170\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"100\" height=\"10\" y=\"35\" x=\"9\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"120\" height=\"10\" y=\"35\" x=\"120\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"90\" height=\"10\" y=\"55\" x=\"5\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"90\" height=\"10\" y=\"55\" x=\"105\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"70\" height=\"10\" y=\"55\" x=\"205\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"30\" height=\"10\" y=\"75\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n",
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, null); })();
    return CodePresetComponent;
}());
CodePresetComponent.ctorParameters = function () { return []; };
var BulletListPresetComponent = /** @class */ (function () {
    function BulletListPresetComponent() {
    }
    BulletListPresetComponent.prototype.ngOnInit = function () {
    };
BulletListPresetComponent.ɵfac = function BulletListPresetComponent_Factory(t) { return new (t || BulletListPresetComponent)(); };
BulletListPresetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BulletListPresetComponent, selectors: [["", "ngx-bullet-list-preset", ""]], attrs: _c8, decls: 8, vars: 0, consts: [["ngx-circle", "", "cx", "7", "cy", "7", "r", "7"], ["ngx-circle", "", "cx", "7", "cy", "37", "r", "7"], ["ngx-circle", "", "cx", "7", "cy", "67", "r", "7"], ["ngx-circle", "", "cx", "7", "cy", "97", "r", "7"], ["ngx-rect", "", "width", "200", "height", "12", "y", "2", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "200", "height", "12", "y", "32", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "200", "height", "12", "y", "62", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "200", "height", "12", "y", "92", "x", "20", "rx", "5", "ry", "5"]], template: function BulletListPresetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "g", 0);
        ɵngcc0.ɵɵelement(1, "g", 1);
        ɵngcc0.ɵɵelement(2, "g", 2);
        ɵngcc0.ɵɵelement(3, "g", 3);
        ɵngcc0.ɵɵelement(4, "g", 4);
        ɵngcc0.ɵɵelement(5, "g", 5);
        ɵngcc0.ɵɵelement(6, "g", 6);
        ɵngcc0.ɵɵelement(7, "g", 7);
    } }, directives: [CircleComponent, RectComponent], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BulletListPresetComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-bullet-list-preset]',
                template: "<svg:g ngx-circle cx=\"7\" cy=\"7\" r=\"7\"></svg:g>\n<svg:g ngx-circle cx=\"7\" cy=\"37\" r=\"7\"></svg:g>\n<svg:g ngx-circle cx=\"7\" cy=\"67\" r=\"7\"></svg:g>\n<svg:g ngx-circle cx=\"7\" cy=\"97\" r=\"7\"></svg:g>\n<svg:g ngx-rect width=\"200\" height=\"12\" y=\"2\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"200\" height=\"12\" y=\"32\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"200\" height=\"12\" y=\"62\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"200\" height=\"12\" y=\"92\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n",
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, null); })();
    return BulletListPresetComponent;
}());
BulletListPresetComponent.ctorParameters = function () { return []; };
var ListPresetComponent = /** @class */ (function () {
    function ListPresetComponent() {
    }
    ListPresetComponent.prototype.ngOnInit = function () {
    };
ListPresetComponent.ɵfac = function ListPresetComponent_Factory(t) { return new (t || ListPresetComponent)(); };
ListPresetComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ListPresetComponent, selectors: [["", "ngx-list-preset", ""]], attrs: _c9, decls: 6, vars: 0, consts: [["ngx-rect", "", "width", "200", "height", "10", "y", "0", "x", "0", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "180", "height", "10", "y", "20", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "160", "height", "10", "y", "40", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "200", "height", "10", "y", "60", "x", "0", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "180", "height", "10", "y", "80", "x", "20", "rx", "5", "ry", "5"], ["ngx-rect", "", "width", "60", "height", "10", "y", "100", "x", "20", "rx", "5", "ry", "5"]], template: function ListPresetComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelement(0, "g", 0);
        ɵngcc0.ɵɵelement(1, "g", 1);
        ɵngcc0.ɵɵelement(2, "g", 2);
        ɵngcc0.ɵɵelement(3, "g", 3);
        ɵngcc0.ɵɵelement(4, "g", 4);
        ɵngcc0.ɵɵelement(5, "g", 5);
    } }, directives: [RectComponent], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ListPresetComponent, [{
        type: core.Component,
        args: [{
                selector: '[ngx-list-preset]',
                template: "<svg:g ngx-rect width=\"200\" height=\"10\" y=\"0\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"180\" height=\"10\" y=\"20\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"160\" height=\"10\" y=\"40\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"200\" height=\"10\" y=\"60\" x=\"0\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"180\" height=\"10\" y=\"80\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n<svg:g ngx-rect width=\"60\" height=\"10\" y=\"100\" x=\"20\" rx=\"5\" ry=\"5\"></svg:g>\n",
                changeDetection: core.ChangeDetectionStrategy.OnPush
            }]
    }], function () { return []; }, null); })();
    return ListPresetComponent;
}());
ListPresetComponent.ctorParameters = function () { return []; };
var NgxContentLoadingModule = /** @class */ (function () {
    function NgxContentLoadingModule() {
    }
NgxContentLoadingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxContentLoadingModule });
NgxContentLoadingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxContentLoadingModule_Factory(t) { return new (t || NgxContentLoadingModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxContentLoadingModule, { declarations: [NgxContentLoadingComponent, RectComponent, CircleComponent, SvgElementComponent, SvgStopComponent, FacebookPresetComponent, InstagramPresetComponent, CodePresetComponent, BulletListPresetComponent, ListPresetComponent], imports: [ɵngcc1.CommonModule], exports: [NgxContentLoadingComponent, RectComponent, CircleComponent, FacebookPresetComponent, InstagramPresetComponent, CodePresetComponent, BulletListPresetComponent, ListPresetComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxContentLoadingModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    NgxContentLoadingComponent,
                    RectComponent,
                    CircleComponent,
                    SvgElementComponent,
                    SvgStopComponent,
                    FacebookPresetComponent,
                    InstagramPresetComponent,
                    CodePresetComponent,
                    BulletListPresetComponent,
                    ListPresetComponent
                ],
                exports: [
                    NgxContentLoadingComponent,
                    RectComponent,
                    CircleComponent,
                    FacebookPresetComponent,
                    InstagramPresetComponent,
                    CodePresetComponent,
                    BulletListPresetComponent,
                    ListPresetComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return NgxContentLoadingModule;
}());
NgxContentLoadingModule.ctorParameters = function () { return []; };

exports.NgxContentLoadingModule = NgxContentLoadingModule;
exports.ɵi = BulletListPresetComponent;
exports.ɵd = CircleComponent;
exports.ɵh = CodePresetComponent;
exports.ɵf = FacebookPresetComponent;
exports.ɵg = InstagramPresetComponent;
exports.ɵj = ListPresetComponent;
exports.ɵa = NgxContentLoadingComponent;
exports.ɵb = RectComponent;
exports.ɵc = SvgElementComponent;
exports.ɵe = SvgStopComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
